Clava-JS/src-api/clava/events/EventListener.ts

```diff
@@ -0,0 +1,141 @@
+import { EventEmitter } from "events";
+import { Event, EventTime } from "./Events.js";
+import ophistory from  "../history/History.js"
+import { DetachOperation, DetachReference, InlineCommentOperation, InsertOperation, RemoveChildrenOperation, ReplaceOperation, SetChildOperation, TypeChangeOperation, ValueOperation } from "../history/Operations.js";
+import { Joinpoint } from "../../Joinpoints.js";
+
+const eventListener = new EventEmitter();
```

Comment:
There should be a global event listener, instantiated by a class in in the lara-framework repo, and other libraries/code register events to that event listener.

Only a small part of this file should be moved to the lara-framework, the part that imports the EventEmitter and instantiates it, it should provide an EventListener class (or LaraEventListener) with a static method that returns the global event listener, and possibly some utility methods (if it makes sense).